Themis/Models/Repository.swift

```diff
@@ -132,6 +135,53 @@ class Node: Hashable, ObservableObject {
     func hash(into hasher: inout Hasher) {
         hasher.combine(path)
     }
+
+    @MainActor
+    public func calculateDiff(templateParticipationId: Int) async {
+        // only calculate once
+        if diffAdded != nil && diffRemoved != nil {
+            return
+        }
+        guard let code = self.code else {
+            return
+        }
+        if templateCode == nil {
+            await fetchTemplateCode(templateParticipationId: templateParticipationId)
+        }
+        guard let tcode = templateCode else {
+            return
```

Comment:
Perhaps we should throw here, because that would be an unexpected condition and something would be wrong. Not sure whether that's necessary though

Reply:
Throwing only makes sense if the user should be notified if the template couldn't be fetched. However we will not want to notify the user about this for every submission he assessess. It just makes more sense to just not calculate/ignore the diff in that case.